server/channels/api4/channel_category_test.go

```diff
@@ -550,6 +551,791 @@ func TestUpdateCategoriesForTeamForUser(t *testing.T) {
+		require.Error(t, err)
+		require.Equal(t, http.StatusForbidden, resp.StatusCode)
+	})
+
+	t.Run("should return error with invalid user id", func(t *testing.T) {
+		categories, _, err := th.Client.GetSidebarCategoriesForTeamForUser(context.Background(), th.BasicUser.Id, th.BasicTeam.Id, "")
+		require.NoError(t, err)
+		require.NotEmpty(t, categories.Categories)
+
+		_, resp, err := th.Client.GetSidebarCategoryForTeamForUser(context.Background(), "invalid_user_id", th.BasicTeam.Id, categories.Categories[0].Id, "")
+		require.Error(t, err)
+		require.Equal(t, http.StatusNotFound, resp.StatusCode)
+	})
+
+	t.Run("should return error with invalid team id", func(t *testing.T) {
+		categories, _, err := th.Client.GetSidebarCategoriesForTeamForUser(context.Background(), th.BasicUser.Id, th.BasicTeam.Id, "")
+		require.NoError(t, err)
+		require.NotEmpty(t, categories.Categories)
+
+		_, resp, err := th.Client.GetSidebarCategoryForTeamForUser(context.Background(), th.BasicUser.Id, "invalid_team_id", categories.Categories[0].Id, "")
+		require.Error(t, err)
+		require.Equal(t, http.StatusNotFound, resp.StatusCode)
+	})
+
+	t.Run("should return error with invalid category id", func(t *testing.T) {
+		_, resp, err := th.Client.GetSidebarCategoryForTeamForUser(context.Background(), th.BasicUser.Id, th.BasicTeam.Id, "invalid_category_id", "")
+		require.Error(t, err)
+		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
+	})
+
+	t.Run("should return error when user is not logged in", func(t *testing.T) {
+		categories, _, err := th.Client.GetSidebarCategoriesForTeamForUser(context.Background(), th.BasicUser.Id, th.BasicTeam.Id, "")
+		require.NoError(t, err)
+		require.NotEmpty(t, categories.Categories)
+
+		client := th.CreateClient()
+		_, resp, err := client.GetSidebarCategoryForTeamForUser(context.Background(), th.BasicUser.Id, th.BasicTeam.Id, categories.Categories[0].Id, "")
+		require.Error(t, err)
+		require.Equal(t, http.StatusUnauthorized, resp.StatusCode)
+	})
+
+	t.Run("should not crash with null input", func(t *testing.T) {
+		require.NotPanics(t, func() {
+			user, client := setupUserForSubtest(t, th)
+			categories, _, err := client.GetSidebarCategoriesForTeamForUser(context.Background(), user.Id, th.BasicTeam.Id, "")
+			require.NoError(t, err)
+			require.NotEmpty(t, categories.Categories)
+
+			route := fmt.Sprintf("/users/%s/teams/%s/channels/categories/%s", user.Id, th.BasicTeam.Id, categories.Categories[0].Id)
+			r, err := client.DoAPIGet(context.Background(), route, "")
+			require.NoError(t, err)
+			closeBody(r)
+		})
+	})
+}
+
+func TestValidateSidebarCategory(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("should validate category with valid channels", func(t *testing.T) {
+		user, _ := setupUserForSubtest(t, th)
+
+		// Create a test context with logger
+		c := &Context{
```

Comment:
This Context is the same between each of these subtests. Could you move it out of the tests instead of repeating it?